authors/views.py

```diff
@@ -161,3 +165,109 @@ def internally_send_friend_request(request, author_id, foreign_author_url):
+            raise exceptions.NotFound
+        # find all author following this author
+        return Author.objects.filter(followings__followee=author)
+
+
+class FollowerDetail(APIView):
+    # TODO permission_classes = [permissions.IsAuthenticatedOrReadOnly]
+
+    @extend_schema(
+        responses=AuthorSerializer(),
+    )
+    def get(self, request, author_id, foreign_author_url):
+        """
+        check if user at the given foreign url is a follower of the local author
+        response: 200 <Author object of the follower>
+        """
+        try:
+            author = Author.objects.get(id=author_id)
+        except:
+            raise exceptions.NotFound
+        return Response(AuthorSerializer(get_object_or_404(
+            Author,
+            followings__followee=author,  # all author following the author
+            url=foreign_author_url  # AND whose url matches param
+        )).data)
+
+    def delete(self, request, author_id, foreign_author_url):
+        """
+        delete a follower by url
+        response: 200 successfully deleted
+        """
+        try:
+            author = Author.objects.get(id=author_id)
+        except:
+            raise exceptions.NotFound("local author is not found")
+
+        try:
+            # the following object for this relationship
+            follower_following = author.followers.get(
+                follower__url=foreign_author_url)
+        except:
+            raise exceptions.NotFound(f"foreign author at {foreign_author_url} is not a follower of the local author")
+
+        follower_following.delete()
+        return Response()
+
+    @extend_schema(
+        examples=[
+            OpenApiExample('A Foreign Author Paylod (Optional)',value={
+                "type": "author",
+                "id": "http://127.0.0.1:8000/author/change-me-123123/",
+                "host": "http://127.0.0.1:8000/",
+                "displayName": "Change Me",
+                "url": "http://127.0.0.1:8000/author/change-me-123123/",
+                "github": "https://github.com/123123123asdafsdfasdfasdfasdf/"
+            })
+        ],
+        request={
+            'application/json': OpenApiTypes.OBJECT
+        },
+    )
+    def put(self, request, author_id, foreign_author_url):
+        """
+        Add a follower (must be authenticated)
+
+        payload: <Author>
+        response: 200 successfully created follow relation
+        """
+        try:
+            author = Author.objects.get(id=author_id)
+        except:
+            raise exceptions.NotFound
+
+        if request.data:
+            follower_serializer = AuthorSerializer(data=request.data)
+        else:
+            # try fetch the foreign user first, upcreate it locally and do it again.
+            # TODO server2server basic auth, refactor into server2server connection pool/service
+            res = requests.get(foreign_author_url)
+            follower_serializer = AuthorSerializer(data=res.text)
```

Comment:
I'm just checking the spec and wondering how should we handle local following requests (local author following another local author)? 

Because when I tested with a local author, it created another author with id as the full url
![image](https://user-images.githubusercontent.com/43459469/137846065-6e8754bd-c1ce-4d70-b67b-e66b63623048.png)

Reply:
yup that should work fine